scripts/test-functional

```diff
@@ -1,44 +1,50 @@
 #!/usr/bin/env node
 
+const path = require('path');
+
 const shell = require('shelljs');
+const tmp = require('tmp');
 
 const config = require('./lib/config');
 const {mochaFunctional} = require('./lib/mocha');
 
 shell.set('-e');
 
 const testProductionMode = process.env.TEST_PRODUCTION_MODE === '1';
+const testLegacyBundling = process.env.TEST_LEGACY_BUNDLING === '1';
 
-let execBuildOptions = {};
 let execMochaOptions = {};
 
+shell.exec('npm run build', testProductionMode ? {
+  env: {
+    ...process.env,
+    NODE_ENV: 'production',
+  },
+} : {});
+
 if (testProductionMode) {
-  execBuildOptions = {
-    env: {
-      ...process.env,
-      NODE_ENV: 'production',
-    },
-  };
+  const destDir = tmp.tmpNameSync();
+  const packageDir = tmp.tmpNameSync();
+  const npmInstallOptions = ['--production'];
+
+  if (testLegacyBundling) {
+    shell.echo('\nTest in "npm legacy bundling mode"');
+    npmInstallOptions.push('--legacy-bundling');
+  }
 
   execMochaOptions = {
     env: {
       ...process.env,
-      TEST_WEB_EXT_BIN: './artifacts/production/bin/web-ext',
+      TEST_WEB_EXT_BIN: path.join(destDir, 'node_modules', 'web-ext', 'bin', 'web-ext'),
```

Comment:
Real production would be `path.join(destDir, 'node_modules', '.bin', 'web-ext')`.

Reply:
yeah, but I had to change it because it is not a nodejs script on windows, I'll leave it as it is.